src/Compilers/CSharp/Portable/Generated/CSharp.Generated.g4

```diff
@@ -242,7 +242,31 @@ accessor_list
   ;
 
 accessor_declaration
-  : attribute_list* modifier* ('get' | 'set' | 'add' | 'remove' | identifier_token) (block | (arrow_expression_clause ';'))
+  : add_accessor_declaration
+  | get_accessor_declaration
+  | remove_accessor_declaration
+  | set_accessor_declaration
+  | unknown_accessor_declaration
```

Comment:
tweaked things based on work i did in VB.  if we have concrete nodes that represent a set of different 'kinds', we now act as if that's an abstract production that points at all the concrete versions.  In VB this is much more critical, as otherwise you have tons of productions that are like `enum_block: enum_statement enum_member* end_block` where `end_block` is then literally a union of ever end statement.

Reply:
looking at the changes ehre, i can see arguments for both directions.  it's far less valuable in C# and i could undo this.  Do you have a preference here @jcouv ?

Reply:
I'm def leaning on the reverting of this bit for C#.  it just doesn't seem worth it to me looking at the diff.

Reply:
I need to look at the code. I don't understand how the `add_accessor_declaration` (`<Kind Name="AddAccessorDeclaration"/>`)  gets associated with the `add` keyword (`<Kind Name="AddKeyword"/>`). That seems dicey.

Reply:
it's literally 1:1 in order.  this is how the normal generators work as well :)

Reply:
See, for example, the generated functions like this in Syntax.xml.Main.Generated.cs:

```c#
    private static SyntaxKind GetAccessorDeclarationKeywordKind(SyntaxKind kind)
    {
      switch (kind)
      {
        case SyntaxKind.GetAccessorDeclaration:
          return SyntaxKind.GetKeyword;
        case SyntaxKind.SetAccessorDeclaration:
          return SyntaxKind.SetKeyword;
        case SyntaxKind.AddAccessorDeclaration:
          return SyntaxKind.AddKeyword;
        case SyntaxKind.RemoveAccessorDeclaration:
          return SyntaxKind.RemoveKeyword;
        case SyntaxKind.UnknownAccessorDeclaration:
          return SyntaxKind.IdentifierToken;
        default:
          throw new ArgumentOutOfRangeException();
      }
    }
```

Reply:
The logic the compiler uses to determine this is simply:

```c#
field.Kinds.Count > 1 && field.Kinds.Count == node.Kinds.Count
```

Reply:
(which is what my check does as well).

Reply:
i've changed my mind.  now i like this in C# :)

Reply:
i changed my mind again and removed this from C# :)